UnityGLTF/Assets/UnityGLTF/Runtime/Scripts/GLTFSceneExporter.cs

```diff
@@ -637,7 +724,108 @@ private static bool IsPrimitive(GameObject gameObject)
 				&& gameObject.transform.localRotation == Quaternion.identity
 				&& gameObject.transform.localScale == Vector3.one
 				&& ContainsValidRenderer(gameObject);
+		}
+		
+		private void ExportSkin(string name, GameObject[] primitives, Node node)
+		{
+			foreach (var prim in primitives)
+			{
+				var smr = prim.GetComponent<SkinnedMeshRenderer>();
+				if (_existedSkins.TryGetValue(smr.rootBone.GetInstanceID(), out SkinId skinId))
+				{
+					node.Skin = skinId;
+					node.Mesh = ExportMesh(prim.name, new GameObject[] { prim });
+					continue;
+				}
+
+				var skin = new Skin
+				{
+					Joints = new List<NodeId>()
+				};
+
+				var mesh = smr.sharedMesh;
+				if (mesh.bindposes.Length != 0)
+				{
+					skin.InverseBindMatrices = ExportAccessor(SchemaExtensions.ConvertMatrix4x4CoordinateSpaceAndCopy(mesh.bindposes));
+				}
+
+				var baseId = _root.Nodes.Count;
+
+				foreach (var bone in smr.bones)
+				{
+					var translation = bone.localPosition;
+					var rotation = bone.localRotation;
+					var scale = bone.localScale;
+					
+					NodeId nodeId = null;
+					if (!_existedNodes.TryGetValue(bone.GetInstanceID(), out nodeId))
+					{
+						var boneNode = new Node
+						{
+							Name = bone.gameObject.name,
+							Translation = new GLTF.Math.Vector3(translation.x, translation.y, translation.z),
+							Rotation = new GLTF.Math.Quaternion(rotation.x, rotation.y, rotation.z, rotation.w),
+							Scale = new GLTF.Math.Vector3(scale.x, scale.y, scale.z),
+						};
+						
+						if (bone.childCount > 0)
+						{
+							boneNode.Children = new List<NodeId>();
+							for (var i = 0; i < bone.childCount; ++i)
+							{
+								var childIndex = Array.IndexOf(smr.bones, bone.GetChild(i));
+								if (-1 == childIndex)
+								{
+									continue;
+								}
+								boneNode.Children.Add(
+									new NodeId
+									{
+										Id = childIndex + baseId,
+										Root = _root
+									}
+								);
+							}
+						}
+						
+						nodeId = new NodeId
+						{
+							Id = _root.Nodes.Count,
+							Root = _root
+						};
+						
+						_root.Nodes.Add(boneNode);
```

Comment:
Generate GLTF nodes is not add to existedNode dict and the _root.Nodes. Would cause duplicated nodes when add gameobject.

It would be more clear to generate GLTF nodes for every gameobject first, then parse components like mesh/skin

Reply:
> It would be more clear to generate GLTF nodes for every gameobject first, then parse components like mesh/skin

Hi, thanks very much! I've fixed the problems you found. I agree with you at this problem, but I don't want to make sweeping changes to the current code structure.

Reply:
> It would be more clear to generate GLTF nodes for every gameobject first, then parse components like mesh/skin

Hi, thanks very much! I've fixed the problems you found. I couldn't agree more with your suggestion, but I don't want to make sweeping changes to the current code structure.